extensions/notification/utils/config.js

```diff
@@ -43,42 +43,45 @@ function validateEntryWithFilter(filterRules, entry) {
 }
 
 /**
- * Validate event against bucket notification configuration event type
- * @param  {Object[]} bnConfigs - Array of bucket specific notification
+ * Validate event against bucket queue configuration event type
+ * @param  {Object[]} bnConfigs - Array of bucket specific queue
  * configurations
  * @param  {string} event - Type of event
- * @return {undefined|Object} A bucket notification configuration if found
+ * @return {undefined|Object[]} Bucket queue configurations if found
  */
 function validateConfigsEvent(bnConfigs, event) {
-    const config = bnConfigs.filter(config => config.events.includes(event));
-    return config.length > 0 ? config[0] : undefined;
+    const configs = bnConfigs.filter(config => config.events.includes(event));
+    return configs.length > 0 ? configs : undefined;
 }
 
 /**
  * Validates an entry from log against bucket notification configurations to see
  * if the entry has to be published. Validations include, bucket specific
  * configuration check, event type check, object name specific filter checks
- * @param  {Object[]} bnConfigs - Array of bucket notifications configuration
+ * @param  {Object} bnConfig - Bucket notification configuration
  * @param  {Object} entry - An entry from the log
  * @return {boolean} true if event qualifies for notification
  */
-function validateEntry(bnConfigs, entry) {
+function validateEntry(bnConfig, entry) {
     const { bucket, type } = entry;
-    // check if the entry belongs to any bucket in the configuration
-    if (bnConfigs.has(bucket)) {
-        const bnConfig = bnConfigs.get(bucket);
-        // check if the event type matches
-        const qConfig = validateConfigsEvent(bnConfig, type);
-        if (qConfig) {
-            // check if there are filters and the object matches it
-            const configFilter = qConfig.filter;
-            if (configFilter && configFilter.filterRules) {
-                return validateEntryWithFilter(configFilter.filterRules, entry);
-            }
-            return true;
-        }
+    const notifConf = bnConfig.notificationConfiguration;
+    // check if the entry belongs to the bucket in the configuration
+    if (bucket !== bnConfig.bucket) {
         return false;
     }
+    // check if the event type matches
+    const qConfigs = validateConfigsEvent(notifConf.queueConfig, type);
+    if (qConfigs) {
+        // check if there are filters for any config and the object's key
+        // satisfies any filter
+        const qFilters
+            = qConfigs.filter(c => c.filterRules && c.filterRules.length > 0);
```

Comment:
`c.filterRules.length > 0` seemed it might be redundant and just the `c.filterRules` might be sufficient alone but I might just be wrong.

Reply:
Yeah, the filter rules are optional when you set bucket notification configuration, may be just checking the length would be correct here, I will change this.